### ISSUE TITLE
Issue when parsing some test google docstrings 

### ISSUE THREAD

**Issue by gcarleo** [ooreilly/mydocstring#2]
@ooreilly thanks for sharing this excellent library! 

I just wanted to mention that I found some issues when parsing some [test Google docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html). 
For example, for this function 

```python
def module_level_function(param1, param2=None, *args, **kwargs):
    """This is an example of a module level function.

    Function parameters should be documented in the ``Args`` section. The name
    of each parameter is required. The type and description of each parameter
    is optional, but should be included if not obvious.

    If \*args or \*\*kwargs are accepted,
    they should be listed as ``*args`` and ``**kwargs``.

    The format for a parameter is::

        name (type): description
            The description may span multiple lines. Following
            lines should be indented. The "(type)" is optional.

            Multiple paragraphs are supported in parameter
            descriptions.

    Args:
        param1 (int): The first parameter.
        param2 (:obj:`str`, optional): The second parameter. Defaults to None.
            Second line of description should be indented.
        *args: Variable length argument list.
        **kwargs: Arbitrary keyword arguments.

    Returns:
        bool: True if successful, False otherwise.

        The return type is optional and may be specified at the beginning of
        the ``Returns`` section followed by a colon.

        The ``Returns`` section may span multiple lines and paragraphs.
        Following lines should be indented to match the first line.

        The ``Returns`` section supports any reStructuredText formatting,
        including literal blocks::

            {
                'param1': param1,
                'param2': param2
            }

    Raises:
        AttributeError: The ``Raises`` section is a list of all exceptions
            that are relevant to the interface.
        ValueError: If `param2` is equal to `param1`.

    """
```

the parser seems to fail when dealing with 
```python
    The format for a parameter is::

        name (type): description
            The description may span multiple lines. Following
            lines should be indented. The "(type)" is optional.

            Multiple paragraphs are supported in parameter
            descriptions.
```

and 
```python
       The ``Returns`` section supports any reStructuredText formatting,
        including literal blocks::

            {
                'param1': param1,
                'param2': param2
            }

```

this is certainly a corner case for users, but might be worth looking into?

**Comment by ooreilly** [comment]
@gcarleo glad to hear that you like my work!

This is an interesting example that you tried. I think I have an idea of what is going wrong. Probably, the simplest solution would be to disable arg parsing unless a recognized header is first detected. Let me elaborate.

If we run with
```
mydocstring google.py module_level_function --json > google.json
```
We can see how the docstring gets parsed and how the different pieces get classified. Here is the first issue 

```
   {
        "args": [
            {
                "description": "description\n    The description may span multiple lines. Following\n    lines should be indented. The \"(type)\" is optional.\n\n    Multiple paragraphs are supported in parameter\n    descriptions.",
                "field": "name",
                "signature": "(type)"
            }
        ],
        "header": "",
        "text": "\nThis is an example of a module level function.\n\nFunction parameters should be documented in the ``Args`` section. The name\nof each parameter is required. The type and description of each parameter\nis optional, but should be included if not obvious.\n\nIf \\*args or \\*\\*kwargs are accepted,\nthey should be listed as ``*args`` and ``**kwargs``.\n\nThe format for a parameter is::\n\n"
    },  
```
Note how the first block of text contains, a blob of text and parameter list. This parameter list gets picked up and parsed. Since `the format for a parameter is::` is not a recognized header. It is rendered as text. What I did in the markdown template file `templates/google_docstring.md` is to first render header, the argument list, and then any text. So we get the following mess in the output for this example: 

```
python
def module_level_function(param1, param2=None, *args, **kwargs):

* **name** (type) : description
    The description may span multiple lines. Following
    lines should be indented. The "(type)" is optional.

    Multiple paragraphs are supported in parameter
    descriptions.

This is an example of a module level function.

Function parameters should be documented in the ``Args`` section. The name
of each parameter is required. The type and description of each parameter
is optional, but should be included if not obvious.

If \*args or \*\*kwargs are accepted,
they should be listed as ``*args`` and ``**kwargs``.

The format for a parameter is::

```

The second issue in the returns section is of similar character. Here, the restructured text block contains text that matches that of the parameter list format. It therefore gets extracted and placed before the text again. Although, here it looks like the parsing is failing for some reason.
input:
```
            {
                'param1': param1,
                'param2': param2
            }    
``` 
output:
```
* param1,
        'param2': param2
    }  
```

I think the simplest solution for now is to avoid parsing any text that matches the parameter list format unless there is known header like `Args` before it.  I can give that solution a go. Let me get back to you.

**Comment by ooreilly** [comment]
This issue has been addressed in #5. Please reopen if it persists.
